### PR TITLE
sim: Enable ubsan in sim:kasan

### PR DESCRIPTION
**Pull request by no1wudi**
## Summary
Change some prototypes of KASAN function to pass build with both KASAN and UBSan enabled.
## Impact
Refactor only
## Testing
sim

### CODE REVIEW

File: mm/kasan/kasan.c [hunk 1]
```diff
@@ -222,188 +222,188 @@ void __asan_handle_no_return(void)
   /* Shut up compiler complaints */
 }
 
-void __asan_report_load_n_noabort(uintptr_t addr, size_t size)
+void __asan_report_load_n_noabort(FAR void *addr, size_t size)
 {
-  kasan_report(addr, size, false);
+  kasan_report((uintptr_t)addr, size, false);
```
Comment (xiaoxiang781216) [inline mm/kasan/kasan.c:227]: let's change the prototype of kasan_report and kasan_set_posion to avoid the cast

File: mm/kasan/kasan.c [hunk 2]
```diff
@@ -107,12 +107,12 @@ static void kasan_report(uintptr_t addr, size_t size, bool is_write)
...
 }
 
-static bool kasan_is_poisoned(uintptr_t addr, size_t size)
+static bool kasan_is_poisoned(FAR void *addr, size_t size)
 {
   FAR uintptr_t *p;
   unsigned int bit;
 
-  p = kasan_mem_to_shadow(addr + size - 1, 1, &bit);
+  p = kasan_mem_to_shadow((uintptr_t)(addr + size - 1), 1, &bit);
```
Comment (xiaoxiang781216) [inline mm/kasan/kasan.c:115]: let's change kasan_mem_to_shadow and kasan_set_poison prototype to FAR void * too

File: mm/kasan/kasan.c [hunk 3]
```diff
@@ -116,7 +117,7 @@ static bool kasan_is_poisoned(uintptr_t addr, size_t size)
   return p && ((*p >> bit) & 1);
 }
 
-static void kasan_set_poison(uintptr_t addr, size_t size, bool poisoned)
+static void kasan_set_poison(FAR void *addr, size_t size, bool poisoned)
```
Comment (xiaoxiang781216) [inline mm/kasan/kasan.c:120]: let's add const and remove the cast from 174 and 179

File: mm/kasan/kasan.c [hunk 4]
```diff
@@ -107,7 +108,7 @@ static void kasan_report(uintptr_t addr, size_t size, bool is_write)
   --recursion;
 }
 
-static bool kasan_is_poisoned(uintptr_t addr, size_t size)
+static bool kasan_is_poisoned(FAR void *addr, size_t size)
```
Comment (xiaoxiang781216) [inline mm/kasan/kasan.c:111]: add const too

File: mm/kasan/kasan.c [hunk 5]
```diff
@@ -93,7 +94,7 @@ static FAR uintptr_t *kasan_mem_to_shadow(uintptr_t addr, size_t size,
   return NULL;
 }
 
-static void kasan_report(uintptr_t addr, size_t size, bool is_write)
+static void kasan_report(FAR void *addr, size_t size, bool is_write)
```
Comment (xiaoxiang781216) [inline mm/kasan/kasan.c:97]: ditto

File: mm/kasan/kasan.c [hunk 6]
```diff
@@ -73,10 +73,11 @@ static FAR struct kasan_region_s *g_region;
  * Private Functions
  ****************************************************************************/
 
-static FAR uintptr_t *kasan_mem_to_shadow(uintptr_t addr, size_t size,
+static FAR uintptr_t *kasan_mem_to_shadow(FAR void *ptr, size_t size,
```
Comment (xiaoxiang781216) [inline mm/kasan/kasan.c:76]: add const

**Review by pkarashchenko — APPROVED** [review]

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**